irods/connection.py

```diff
@@ -437,13 +456,31 @@ def _login_gsi(self):
 
     def _login_pam(self):
 
-        time_to_live_in_seconds = 60
-
-        pam_password = PAM_PW_ESC_PATTERN.sub(lambda m: '\\'+m.group(1), self.account.password)
+        import irods.client_configuration as cfg
+        inline_password = (self.account.authentication_scheme == self.account._original_authentication_scheme)
+        time_to_live_in_hours = 60
+        new_pam_password = PAM_PW_ESC_PATTERN.sub(lambda m: '\\'+m.group(1), self.account.password)
+        if not inline_password:
+            # Login in using PAM password from .irodsA
+            try:
+                self._login_native()
+            except (ex.CAT_PASSWORD_EXPIRED, ex.CAT_INVALID_USER, ex.CAT_INVALID_AUTHENTICATION):
+                time_to_live_in_hours = cfg.legacy_auth.pam.time_to_live_in_hours # default is 60.  TODO: document setting.
+                if cfg.legacy_auth.pam.auto_renew_password:
+                    new_pam_password = cfg.legacy_auth.pam.auto_renew_password
```

Comment:
It will in fact be a string value giving the password value, since that information is needed for the algorithm and not available by any other means.  Yes, I'll definitely try to think of a better name.

Reply:
Done: changed the setting name to `password_for_auto_renew`